cg/meta/clean/clean_flow_cells.py

```diff
@@ -0,0 +1,134 @@
+            LOG.error(f"Flow cell with path {self.flow_cell.path} not removed: {str(error)}")
+            return is_error_raised
+
+    def can_flow_cell_directory_be_deleted(self) -> bool:
+        """Determine whether a flow cell directory can be deleted."""
+        return all(
+            [
+                self.is_directory_older_than_days_old(),
+                self.is_flow_cell_in_statusdb(),
+                self.is_flow_cell_backed_up(),
+                self.has_sequencing_metrics(),
+                self.has_fastq_files_in_housekeeper(),
+                self.has_spring_files_in_housekeeper(),
+                self.has_sample_sheet_in_housekeeper(),
+            ]
+        )
+
+    def is_directory_older_than_days_old(self) -> bool:
+        """Check if a given directory is older than specified number of days.
+        The check is performed by comparing whether the directory creation time stamp in seconds
+        is greater than the current time minus 21 days in seconds.
+        """
+        dir_creation_time_stamp: float = get_creation_time_stamp(self.flow_cell.path)
+
+        return bool(dir_creation_time_stamp < self.current_time - TWENTY_ONE_DAYS_IN_SECONDS)
+
+    def get_flow_cell_from_status_db(self) -> Flowcell:
+        """Get the flow cell entry from statusDB.
+        Raises:
+              ValueError if the flow cell is not found in statusDB
+        """
+        flow_cell: Flowcell = self.status_db.get_flow_cell_by_name(self.flow_cell.id)
+        if not flow_cell:
+            raise ValueError(f"Flow cell {flow_cell.id} not found in statusDB.")
+        return flow_cell
+
+    def is_flow_cell_in_statusdb(self) -> bool:
+        """Check if flow cell is in statusdb."""
+        return bool(self.get_flow_cell_from_status_db())
+
+    def is_flow_cell_backed_up(self) -> bool:
+        """Check if flow cell is backed up on PDC."""
+        return bool(self.get_flow_cell_from_status_db().has_backup)
+
+    def get_sequencing_metrics_for_flow_cell(self):
+        """Get the SampleLaneSequencingMetrics entries for a flow cell."""
+        metrics: List[
+            SampleLaneSequencingMetrics
+        ] = self.status_db.get_sample_lane_sequencing_metrics_by_flow_cell_name(self.flow_cell.id)
+        if not metrics:
+            raise ValueError(
+                f"No SampleLaneSequencingMetrics found for flow cell {self.flow_cell.id} in statusDB"
+            )
+        return metrics
+
+    def has_sequencing_metrics(self):
+        """Check if a flow cell has entries in the SampleLaneSequencingMetrics table."""
+        return bool(self.get_sequencing_metrics_for_flow_cell())
+
+    def has_sample_sheet_in_housekeeper(self) -> bool:
+        """Check if the flow cell has a sample sheet in housekeeper."""
+        return bool(self.flow_cell.get_sample_sheet_path_hk())
+
+    def get_files_for_flow_cell_bundle(self, tag: str):
+        """Get the files with a specific tag from housekeeper for the flow cell bundle."""
```

Comment:
```suggestion
        """Get the files with a specific tag from Housekeeper for the flow cell bundle."""
```

Reply:
Do we want a "Raises" here as well? Can't quite recall if we decided on that being a standard now.

Reply:
The housekeeper function that is called does that. So this function here does not raise anything directly.